db/db.go

```diff
@@ -0,0 +1,472 @@
+//
+//   c := NewConnector(url, database, connection)
+//
+func NewConnector(url string, database string, collection string) Connector {
+	return &mongo{
+		url:        url,
+		database:   database,
+		collection: collection,
+	}
+}
+
+// Connect creates a session to db. All queries to DB will copy this session to
+// make a request. When you are finished, you must call Disconnect to close the session.
+//
+//   c := NewConnector(url, database, connection)
+//   err := c.Connect()
+//
+// Relevant documentation:
+//
+//     https://godoc.org/gopkg.in/mgo.v2#Dial
+//
+func (m *mongo) Connect() error {
+	session, err := mgo.Dial(m.url)
+	if err != nil {
+		// Returning raw, low-level error since this method only does one thing.
+		return err
+	}
+
+	m.session = session
+	return nil
+}
+
+// Disconnect must be called when finished with db connector.
+//
+//   c := NewConnector(url, database, connection)
+//   _ := c.Connect()
+//   defer c.Disconnect()
+//
+// Relevant documentation:
+//
+//     https://godoc.org/gopkg.in/mgo.v2#Session.Close
+//
+func (m *mongo) Disconnect() {
+	m.session.Close()
+}
+
+// CreateEntities inserts many Entity into DB. If succesful, returns slice of IDs. Otherwise, returns an error.
+//
+//   entities:= []db.Entity{
+//     db.Entity{"x": 1},
+//     db.Entity{"y": 2},
+//   }
+//   ids, err := c.CreateEntities(testData)
+//   if err != nil {
+//     if _, ok := err.(ErrCreate); ok {
+//       // db error
+//     } else {
+//       // uncaught error
+//     }
+//   }
+//
+// Relevant documentation:
+//
+//     https://godoc.org/gopkg.in/mgo.v2#Collection.Insert
+//     https://docs.mongodb.com/manual/reference/bson-types/#objectid
+//
+func (m *mongo) CreateEntities(entities []Entity) ([]string, error) {
+	s := m.session.Copy()
+	defer s.Close()
+
+	c := m.session.DB(m.database).C(m.collection)
+
+	// A slice of IDs we generate to insert along with entities into DB
+	objectIDs := make([]string, len(entities))
+
+	// Translate []Entity to []interface{}, as that's what Insert method expects.
+	entitiesToInsert := make([]interface{}, len(entities))
+
+	for i, _ := range entities {
+		// Mgo driver relies on Mongo returning the ID. However, Mongo does not
+		// return that ID, and the driver does not either. To avoid doing a read
+		// after insert, we generate the ID ourselves and then insert the doc with
+		// it.
+		//
+		// We choose to store and return a string because the methods that manage
+		// single entities expect type string for the id argument.
+		objectID := bson.NewObjectId().String()
+
+		// Add ID to slice to return later
+		objectIDs[i] = objectID
+
+		// Update entity to include "_id" field with newly generated id
+		entities[i]["_id"] = objectID
+
+		// Add updated entity to slice of entities that we will insert to DB
+		entitiesToInsert[i] = entities[i]
+	}
+
+	err := c.Insert(entitiesToInsert...)
+	if err != nil {
+		return nil, ErrCreate{DbError: err}
```

Comment:
Partial success? If we insert 6 entities and 5 work and 6th fails, how does caller know or handle this `mgo.Insert` isn't clear on how this works, either, so we need to try it and see and make it explicit here. Being explicit about what got stored, updated, deleted (or what didn't) is mission-critical for this pkg.